### PR TITLE
re-export async-nats, bump async-nats to 0.27.1

### PR DESCRIPTION
**Pull request by stevelr**
re-exported async-nats 
- downstream crates (such as some capability providers) that depend on async-nats directly and also wasmbus_rpc::RpcClient, that get build errors for conflicting versions of async-nats, may be able to fix builds by using `wasmbus_rpc::async_nats::{Client, Message}`from this crate.
- wasmbus-rpc clients are not required to use the re-exported async-nats, and dependent crates are encouraged to use async-nats directly, if it works, rather than through the re-export. `RpcClient::new` takes a parameter of type `async_nats::Client`, and the async_nats crate versions need to match. Avoiding the re-export may require a more significant refactor, or waiting for rust to resolve the [public-private dependency problem](https://github.com/rust-lang/rust/issues/44663 and https://rust-lang.github.io/rfcs/1977-public-private-dependencies.html) . 
Signed-off-by: stevelr <steve@cosmonic.com>

### CODE REVIEW

**Review by brooksmtownsend — COMMENTED** [review]
I like this PR, I'm just concerned that this is more than a patch bump. Will we get issues where the user is supplying an `async_nats::Client` and we expect a `wasmbus_rpc::async_nats::Client`?

**Comment by stevelr** [comment]
> I like this PR, I'm just concerned that this is more than a patch bump. Will we get issues where the user is supplying an `async_nats::Client` and we expect a `wasmbus_rpc::async_nats::Client`?

If we were going to get an error (without this change), we'll still get the exact same error - the additional export doesn't change that. To fix it though, with this change in place, the user can change `async_nats` to `wasmbus_rpc::async_nats` (and remove async_nats from Cargo.toml) and compile should work again.

File: codegen/src/codegen_rust.rs [hunk 1]
```diff
@@ -1212,7 +1212,7 @@ impl<'model> RustCodeGen<'model> {
         writeln!(
             w,
             r#"/// client for sending {} messages
-              #[derive(Debug)]
+              #[derive(Clone, Debug)]
```
Comment (thomastaylor312) [inline codegen/src/codegen_rust.rs:1215]: Isn't this going to cause build failures? `T` isn't clone, so it can't derive

**Review by thomastaylor312 — CHANGES_REQUESTED** [review]
One small blocking change

File: codegen/src/codegen_rust.rs (hunk 1 above)
Comment (stevelr) [reply codegen/src/codegen_rust.rs:1215]: fixed by merging feat/smithy-bindgen branch where I had already applied this fix

**Comment by thomastaylor312** [comment]
@stevelr I think we're ready to go, but would you mind squashing down the 18 commits into few commits for history sake?

File: codegen/src/codegen_rust.rs (hunk 1 above)
Comment (thomastaylor312) [reply codegen/src/codegen_rust.rs:1215]: I'm still seeing `T` below only have a `Transport` bound rather than `Transport + Clone`

Comment (thomastaylor312) [reply codegen/src/codegen_rust.rs:1215]: Oh wait, I found where it is at. Transport must implement Clone

**Review by thomastaylor312 — APPROVED** [review]
Approving, but please squash down the commits if possible before merging

**Pull request merged**